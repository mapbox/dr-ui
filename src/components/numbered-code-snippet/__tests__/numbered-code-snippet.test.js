import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './numbered-code-snippet-test-cases.js';

describe('numbered-code-snippet', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe(testCases.noCollapse.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.noCollapse;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe(testCases.firstLine.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.firstLine;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe(testCases.lastLine.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.lastLine;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
