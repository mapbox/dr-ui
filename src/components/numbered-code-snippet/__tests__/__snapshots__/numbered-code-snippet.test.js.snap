// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`numbered-code-snippet Basic renders as expected 1`] = `
<div
  className="relative round z0 scroll-styled scroll-auto"
  style={
    Object {
      "backgroundColor": "#f4f7fb",
      "maxHeight": 450,
    }
  }
>
  <pre
    className="my-neg12 py12 ml-neg12 pl12 mobile-snippet"
  >
    <code
      className="px0 hljs"
    >
      <div
        className="relative z2"
        data-chunk-code="chunk-0"
      >
        <div
          className="pr12 opacity50 bg-darken10"
          style={
            Object {
              "paddingLeft": 12,
            }
          }
        >
          <div
            className="absolute fl color-gray align-r pr6 w36"
            data-content={1}
            style={
              Object {
                "marginLeft": -12,
                "marginTop": -0,
              }
            }
            unselectable="on"
          />
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<span class=\\"token comment\\">// 'style' in this case refers to an MGLStyle object.</span>",
              }
            }
            style={
              Object {
                "marginLeft": 43.349999999999994,
                "textIndent": -14.45,
              }
            }
          />
        </div>
      </div>
      <div
        className="relative z2"
        data-chunk-code="chunk-1"
      >
        <div
          className="pr12 py3"
          style={
            Object {
              "paddingLeft": 12,
            }
          }
        >
          <div
            className="absolute fl color-gray align-r pr6 w30 py3"
            data-content={2}
            style={
              Object {
                "marginLeft": -6,
                "marginTop": -3,
              }
            }
            unselectable="on"
          />
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<span class=\\"token keyword\\">let</span> layer <span class=\\"token operator\\">=</span> style<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">layer</span><span class=\\"token punctuation\\">(</span>withIdentifier<span class=\\"token punctuation\\">:</span> <span class=\\"token string\\">\\"place-city-sm\\"</span><span class=\\"token punctuation\\">)</span> <span class=\\"token keyword\\">as</span><span class=\\"token operator\\">!</span> <span class=\\"token builtin\\">MGLSymbolStyleLayer</span>",
              }
            }
            style={
              Object {
                "marginLeft": 43.349999999999994,
                "textIndent": -14.45,
              }
            }
          />
        </div>
        <div
          className="pr12 py3"
          style={
            Object {
              "paddingLeft": 12,
            }
          }
        >
          <div
            className="absolute fl color-gray align-r pr6 w30 py3"
            data-content={3}
            style={
              Object {
                "marginLeft": -6,
                "marginTop": -3,
              }
            }
            unselectable="on"
          />
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<span class=\\"token keyword\\">let</span> spanish <span class=\\"token operator\\">=</span> <span class=\\"token function\\">Locale</span><span class=\\"token punctuation\\">(</span>identifier<span class=\\"token punctuation\\">:</span> <span class=\\"token string\\">\\"es\\"</span><span class=\\"token punctuation\\">)</span>",
              }
            }
            style={
              Object {
                "marginLeft": 43.349999999999994,
                "textIndent": -14.45,
              }
            }
          />
        </div>
      </div>
      <div
        className="relative z2"
        data-chunk-code="chunk-2"
      >
        <div
          className="pr12 opacity50 bg-darken10"
          style={
            Object {
              "paddingLeft": 12,
            }
          }
        >
          <div
            className="absolute fl color-gray align-r pr6 w36"
            data-content={4}
            style={
              Object {
                "marginLeft": -12,
                "marginTop": -0,
              }
            }
            unselectable="on"
          />
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "layer<span class=\\"token punctuation\\">.</span>text <span class=\\"token operator\\">=</span> layer<span class=\\"token punctuation\\">.</span>text<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">mgl_expressionLocalized</span><span class=\\"token punctuation\\">(</span>into<span class=\\"token punctuation\\">:</span> spanish<span class=\\"token punctuation\\">)</span>",
              }
            }
            style={
              Object {
                "marginLeft": 43.349999999999994,
                "textIndent": -14.45,
              }
            }
          />
        </div>
      </div>
    </code>
  </pre>
  <div
    className="bg-lighten50 absolute left right"
    data-chunk-overlay="chunk-1"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <div
      className="bg-blue h-full w6"
    />
  </div>
  <div
    className="absolute z3 right mr3 color-white"
    data-chunk-copy="chunk-1"
    style={
      Object {
        "opacity": 0,
        "transition": "opacity 300ms linear",
      }
    }
  />
</div>
`;
